simulation/behavior_tree_plugin/src/vehicle/follow_trajectory_sequence/polyline_trajectory_follower.cpp

```diff
@@ -0,0 +1,647 @@
+                << std::endl;
+
+      std::cout << "desired_speed "
+                << "== speed + std::clamp(desired_acceleration, min_acceleration, max_acceleration) * step_time "
+                << "== " << speed << " + std::clamp(" << desired_acceleration << ", " << min_acceleration << ", " << max_acceleration << ") * " << step_time << " "
+                << "== " << speed << " + " << std::clamp(desired_acceleration, min_acceleration, max_acceleration) << " * " << step_time << " "
+                << "== " << speed << " + " << std::clamp(desired_acceleration, min_acceleration, max_acceleration) * step_time << " "
+                << "== " << desired_speed
+                << std::endl;
+
+      std::cout << "distance_to_front_waypoint "
+                << "== " << distance_to_front_waypoint
+                << std::endl;
+
+      std::cout << "remaining_time_to_arrival_to_front_waypoint "
+                << "== " << remaining_time_to_arrival_to_front_waypoint.value()
+                << std::endl;
+
+      std::cout << "distance "
+                << "== " << distance
+                << std::endl;
+
+      std::cout << "remaining_time "
+                << "== " << remaining_time
+                << std::endl;
+
+      std::cout << "remaining_time_to_arrival_to_front_waypoint "
+                << "("
+                << "== distance_to_front_waypoint / desired_speed "
+                << "== " << distance_to_front_waypoint << " / " << desired_speed << " "
+                << "== " << remaining_time_to_arrival_to_front_waypoint.value()
+                << ")"
+                << std::endl;
+
+      std::cout << "arrive during this frame? "
+                << "== remaining_time_to_arrival_to_front_waypoint < step_time "
+                << "== " << remaining_time_to_arrival_to_front_waypoint.value() << " < " << step_time << " "
+                << "== " << math::arithmetic::isDefinitelyLessThan(remaining_time_to_arrival_to_front_waypoint.value(), step_time)
+                << std::endl;
+
+      std::cout << "not too early? "
+                << "== std::isnan(remaining_time_to_front_waypoint) or remaining_time_to_front_waypoint < remaining_time_to_arrival_to_front_waypoint + step_time "
+                << "== std::isnan(" << remaining_time_to_front_waypoint << ") or " << remaining_time_to_front_waypoint << " < " << remaining_time_to_arrival_to_front_waypoint.value() << " + " << step_time << " "
+                << "== " << std::isnan(remaining_time_to_front_waypoint) << " or " << math::arithmetic::isDefinitelyLessThan(remaining_time_to_front_waypoint, remaining_time_to_arrival_to_front_waypoint.value() + step_time) << " "
+                << "== " << (std::isnan(remaining_time_to_front_waypoint) or math::arithmetic::isDefinitelyLessThan(remaining_time_to_front_waypoint, remaining_time_to_arrival_to_front_waypoint.value() + step_time))
+                << std::endl;
+    // clang-format on
+  }
```

Comment:
zobacz ile bedzie przekazywania argumentow jakby mialo to zostac metoda, ale bardzo chcialbym to wyciac stad